mysql-test/main/outfile_fifo.test

```diff
@@ -0,0 +1,17 @@
+#
+# MDEV-32061: Allow INTO OUTFILE to writo to a fifo
+#
+--disable_warnings
+DROP TABLE IF EXISTS t1;
+--enable_warnings
+CREATE TABLE t1 (val1 VARCHAR(255), val2 VARCHAR(255));
+INSERT INTO t1 VALUES('hello', 'there');
+SELECT * FROM t1;
+SELECT * INTO OUTFILE '../../tmp/fifo' FROM t1;
+TRUNCATE TABLE t1;
+SELECT SLEEP(.25); #To make sure the background "cat" process has a chance to read from the fifo and write to the text file
```

Comment:
Sleep statements are a bit fragile:
concept from `mysql-test/include/wait_until_connected_again.inc`. I haven't looked up the error code.
```
--disable_result_log
--disable_query_log
let $counter= 100;
let $mysql_errno= 9999;
while ($mysql_errno)
{
  --error 0,ER_NO_SUCH_FILE
  LOAD DATA INFILE '../../tmp/fifo_data.txt' INTO TABLE t1;
 
  dec $counter;
  if (!$counter)
  {
    --die Out of time for file to appear
  }
    if ($mysql_errno > 0)
  {
    --sleep 0.1
  }
}
--enable_query_log
--enable_result_log
```

Reply:
grooverdan, thanks for all the feedback.  I'll be making changes tomorrow.

This particular one I'm not sure will work, because the `cat` command creates the output file immediately.  It sits empty until the data arrives in the fifo, at which time it starts being written to the file.

We might need to wait until the file is non-empty.  Even that assumes that the file will be written atomically and we won't catch it in progress, which is /probably/ the case.

Even if we find a way to test for non-empty, I'm not sure it's better than just sleeping for plenty of time.

Reply:
Maybe we could loop until observing the same non-zero file size twice.  That should be sufficient.

Reply:
yep. that sounds all right.

Reply:
This was really tricky, because I couldn't find any way to get the size of a file, or really to get anything about a file into a variable rather than printing its contents.

The trick ended up being to put " && touch done.txt" into the shell script.  Then I could detect the presence of done.txt.

I wasn't able to find a nice name for error 13, though.  Maybe it exists.